sharedbuffers/mapped_struct.py

```diff
@@ -503,6 +521,62 @@ def unpack_from(cls, buf, offs, idmap = None):
             idmap[offs] = rv
         return rv
 
+class mapped_decimal(Decimal):
+    @classmethod
+    @cython.locals(offs = cython.longlong, implicit_offs = cython.longlong, exponent = cython.longlong, sign = cython.uchar)
+    def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0, packer = struct.Struct('=Bq')):
```

Comment:
Usually we place the packer and other Structs as class attributes.

Also... do you really need 64 bits for the exponent? Also those 64 bits just after a byte will have a lot of padding.